src/stdune/path.ml

```diff
@@ -512,15 +557,82 @@ end = struct
                 ["t", to_sexp t]
     | Some parent -> parent
 
-  let of_local t = t
+  module Fix_root (Root : sig type w end) = struct
```

Comment:
Why do we need this functor?

Reply:
It's to be able to have the types Set.t and Map.t whose elements are `w Local_gen.t` for a fixed `w`.